### DISCUSSION TITLE
v1.4.4

### DISCUSSION THREAD

**Discussion by fiammybe** [ImpressCMS/impresscms#1173] [General]
## What's Changed
* Applying the filters to inner elements of arrays by @skenow in https://github.com/ImpressCMS/impresscms/pull/1162
* Additional input filtering - mailusers, findusers, checkVarArray inner elements by @skenow in https://github.com/ImpressCMS/impresscms/pull/1163
* Filtering updates for blocksadmin and mailusers by @skenow in https://github.com/ImpressCMS/impresscms/pull/1164
* Prepare 1.4.4 by @fiammybe in https://github.com/ImpressCMS/impresscms/pull/1171

**Full Changelog**: https://github.com/ImpressCMS/impresscms/compare/v1.4.3...v1.4.4

<hr /><em>This discussion was created from the release <a href='https://github.com/ImpressCMS/impresscms/releases/tag/v1.4.4'>v1.4.4</a>.</em>

**Comment by fiammybe** [comment]
Looks like a change crept in the 1.4.4 code that breaks it when running on really old PHP versions (5.6), as discovered by our partners at softaculous. 
Given that this is still a supported PHP version for the 1.4 branch in order to let people migrate their old sites (mostly running on old PHP versions) to newer versions, we will release 1.4.5 tomorrow as the last version to support PHP < 7.2 with a fix for this.

**Reply by skenow** [reply]
array_fill_keys has been around since PHP 5.2. In some cases, it has problems when the second argument is an array, as it is in our case. It works in my environment, which is PHP 5.6. It may not be a version issue, rather an environment issue.

**Reply by fiammybe** [reply]
I tested it on my windows Xampp 5.6, and I have no issues there either. So that's 2 totally different environments. What exact version did you try it on? 5.6.x? That might be an indication.

**Reply by skenow** [reply]
My local environment is PHP Version 5.6.40-57+ubuntu20.04.1+deb.sury.org+1. A couple of online servers are also 5.6.40.  A test install on one was only at 1.4.3rc, but I applied the change to checkVarArray and did not get any errors.
